### ISSUE TITLE
GPU acceleration

### ISSUE THREAD

**Issue by jaredmontoya** [artemsen/swayimg#175]
It's almost necessary to have GPU acceleration in the current day and age. Without it viewing big images will always be painful, it's just a matter of how big they can get before you feel it.
Thank you in advance!

**Comment by artemsen** [comment]
The only problem is that I couldn't find a way to draw GL texture (image) on a transparent window.
Not now, but maybe I'll return to this issue a little later =)

**Comment by qaako** [comment]
I don't know anything about anything, but I have heard on IRC before, concerning the topic of why feh only uses 15MB per image and there are no lightweight image viewers that support native wayland, that loading a mesa context automatically creates 100MB RSS usage. This makes a lot of sense of why imv is such a memory hog.

If you do add GPU acceleration, I hope it doesn't increase resource usage, because I'd rather render an image 10x slower than even increase RAM usage by 25%. I like to keep 20+ images open and the RAM usage multiplies quickly with most image viewers.

It would be a nice feature if it doesn't have caveats, but this image viewer being the most lightweight is my favorite thing about it. If someone is after cycling through a gallery at maximum rendering speed, there are plenty of heavier image viewers to do the job. Personally, I think Eye of Gnome or LXIMage-QT would accel for that purpose (using a single main window and cycling through gallery) But there are loads of alternatives for this use case. Swayimg is awesome for keeping tons of windows open with a low resource footprint... And it's the only app of its kind on Wayland. I know I'm just one user here, but I would really hate to see this change.

**Comment by jaredmontoya** [comment]
Well, It's not about scrolling through a gallery at maximum speed, it's about viewing 8k images and not having fps drop to 5 while it's moving.

**Comment by diniamo** [comment]
I think there is no real point being made here either, if GPU acceleration were to be implemented, it would be behind a build flag, and a toggle.

**Comment by artemsen** [comment]
To be honest, I don't see any pressing need for this feature.
GPU support in this case will save RAM by using video memory and will provide the ability to smoothly drag with the mouse.
If you don't have a discrete GPU with dedicated VRAM, then the first advantage is questionable.
The smooth drag issue has been partially resolved in version 2.5, I would say "completely resolved" if you turn off antialiasing.

I just tried opening a 30720 x 17280 pixel image, and the current version of swayimg handles it just fine. In fact, the image size only matters when loading, but the rendering speed depends entirely on the window size.

**Comment by diniamo** [comment]
Well yes, there is the issue, I would like to have antialiasing enabled. This is only a slight annoyance though, whatever was done in 2.5 works really well, compared to before.

**Comment by diniamo** [comment]
Nice, seems to make antialiasing usable. Thanks

**Comment by 20NE** [comment]
Yep, with GPU support, rendering blur window background might be much smoother.
